hazelcast/src/test/java/com/hazelcast/map/EntryProcessorOffloadableTest.java

```diff
@@ -867,17 +875,18 @@ public void run() {
                         if (lastbeat != newlastbeat) {
                             lastbeat = newlastbeat;
                             heartBeatTimestamps.add(newlastbeat);
+                            exitLatch.countDown();
```

Comment:
If there are heartbeats from some other operations that come before `TimeConsumingOffloadableTask` starts then the task may finish too early (before "its" heartbeat is registered). As a result, the final assertion in the test may fail.

Reply:
Valid point. I hardened the test on that side a bit, have another look.
Regarding the opposite scenario, I am not concerned that much for the off-loadable endTime (HBs coming after it finished) because that would be a valid test failure.